cmd/root/root.go

```diff
@@ -151,5 +161,40 @@ Stack Trace:
 		}
 	}
 
+	exitCode := 0
+	if err != nil {
+		exitCode = 1
+	}
+
+	uploadTelemetry(cmd.Context(), commandString(cmd), startTime, exitCode)
 	return err
 }
+
+func uploadTelemetry(ctx context.Context, cmdStr string, startTime time.Time, exitCode int) {
```

Comment:
Is there a reason to keep this function here versus in the telemetry package?

Reply:
We do need the `HasLogs` check because not all commands would necessarily have a auth configured, and `ConfigUsed(ctx)` might panic. 

I however moved the other pieces of logic to the `telemetry.Upload` function.